components/ScrollToTopButton/ScrollToTopButton.tsx

```diff
@@ -1,22 +1,46 @@
 import classNames from 'classnames';
+import { useEffect, useState } from 'react';
 
 import { IconCaret } from '@/icons';
 
 import Button from '../Button';
 
 import styles from './ScrollToTopButton.module.scss';
 
-interface Props {
-    isVisible: boolean;
-    onClick: () => void;
-}
+function ScrollToTopButton() {
+    const [isScrollToTopVisible, setIsScrollToTopVisible] = useState(false);
+
+    useEffect(() => {
+        function scrollListener() {
+            if (document.body.scrollTop > 300 || document.documentElement.scrollTop > 300) {
+                setIsScrollToTopVisible(true);
+            } else {
+                setIsScrollToTopVisible(false);
+            }
```

Comment:
This can be made simpler:
```suggestion
                setIsScrollToTopVisible(document.body.scrollTop > 300 || document.documentElement.scrollTop > 300);
```

Reply:
Also better to move the `300` value to a constant

Reply:
updated!